packages/mobile/src/send/SendAmount/SendAmount.test.tsx

```diff
@@ -206,6 +206,47 @@ describe('SendAmount', () => {
       expect(getElementText(getByTestId('InputAmount'))).toEqual('22.85')
     })
 
+    it('max button maxes out the token value only and displays the correct number of decimal places', () => {
+      const store = createMockStore({
+        ...storeData,
+        localCurrency: {
+          preferredCurrencyCode: LocalCurrencyCode.USD,
+          fetchedCurrencyCode: LocalCurrencyCode.USD,
+          exchangeRates: { [Currency.Dollar]: '1' },
+        },
+        tokens: {
+          tokenBalances: {
+            [mockCusdAddress]: {
+              address: mockCusdAddress,
+              symbol: 'cUSD',
+              usdPrice: '1',
+              balance: '22.85789012',
+              isCoreToken: true,
+              priceFetchedAt: Date.now(),
+            },
+          },
+        },
+      })
+      const { getByTestId } = render(
+        <Provider store={store}>
+          <SendAmount {...mockScreenProps()} />
+        </Provider>
+      )
+
+      fireEvent.press(getByTestId('MaxButton'))
+
+      // The value expected here is |balance| - the send fee (1 cUSD)
+      expect(getElementText(getByTestId('InputAmountContainer'))).toEqual('$21.85')
```

Comment:
i suppose we round down so that we don't give the illusion that we are sending more money than the user has? but then we display 21.86cUSD below? i'm not sure what the desired behaviour is to be honest, but we do use 2 different ways to format the 2 lines (i.e. on the input line, we round to 2 decimal but on the line below, we round to 2 significant figures)

![Screenshot 2022-02-03 at 17 41 44](https://user-images.githubusercontent.com/20150449/152392222-fa55c103-e7a2-40b6-8ace-551002fb7b93.png)
![Screenshot 2022-02-03 at 17 41 53](https://user-images.githubusercontent.com/20150449/152392225-69a431ca-d420-4864-a6ea-591d734f07a8.png)

Reply:
i made an executive decision to format the numbers consistently with the send confirmation screen. i just don't think it makes sense to show something like $21.85 on this screen then $21.86 on the send confirmation screen :/